Update RBAC task for Istio 1.0.

yangminzhu opened the pull request:
For istio/istio#6377

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio.github.io/pull/1789#" title="Author self-approved">yangminzhu</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **sdake**

Assign the PR to them by writing `/assign @sdake` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[content/OWNERS](https://github.com/istio/istio.github.io/blob/master/content/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["sdake"]} -->

geeknoid commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -1,29 +1,26 @@
 ---
-title: Role-Based Access Control
+title: Authorization Policy
```
I'm very worried that we're using "policy" in the security space now.

Policy and Telemetry refers to Mixer's domain and the docs are structured that way. Now we'll have Policy and Telemetry sedtions, along with other *Policy* things elsewhere. Very confusing.

We could just call this page "Authorization" I think.

liminw replied:
In the original doc, I call it "Istio Authorization". Let's use that name instead.

yangminzhu replied:
Done.

liminw commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -1,29 +1,26 @@
-RBAC from [Istio RBAC concept page](/docs/concepts/security/#role-based-access-control-rbac).
+This guide covers the main tasks you might need to perform to set up Istio authorization, also known
+as Istio Role Based Access Control (RBAC), for services in an Istio mesh. You can read more in
+[Istio authorization](/docs/concepts/security/#role-based-access-control-rbac).
```
"and get started with a basic tutorial in Istio Security Basics."

yangminzhu replied:
Done, it would be better to link to the page, do you mean this page (https://preliminary.istio.io/docs/tasks/security/authn-policy/) or another page?

liminw replied:
There will be a security CUJ page, which we can link to later.

yangminzhu replied:
Great!

liminw commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -1,29 +1,26 @@
+* Understand [Istio authorization](/docs/concepts/security/#role-based-access-control-rbac) concepts.
+
+* Have set up Istio on Kubernetes **with authentication enabled** by following the instructions in the
   [quick start](/docs/setup/kubernetes/quick-start/).
```
Istio Authentication requires mutual TLS to work. Authentication should be enabled in the installation steps.

yangminzhu replied:
Done. Just curious I thought we could also use JWT-based authentication and in this case mTLS is optional?

liminw replied:
How about change it to: This tutorial requires mutual TLS to work. Mutual TLS authentication should be enabled in the installation steps.

yangminzhu replied:
Done.

geeknoid commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -1,29 +1,28 @@
 
-This task shows how to set up role-based access control (RBAC) for services in Istio mesh. You can read more about Istio
-RBAC from [Istio RBAC concept page](/docs/concepts/security/#role-based-access-control-rbac).
+This guide covers the main tasks you might need to perform to set up Istio authorization, also known
```
guide -> task

yangminzhu replied:
Done.

geeknoid commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -1,29 +1,28 @@
+This guide covers the main tasks you might need to perform to set up Istio authorization, also known
+as Istio Role Based Access Control (RBAC), for services in an Istio mesh. You can read more in
+[Istio authorization](/docs/concepts/security/#role-based-access-control-rbac) and get started with
+a basic tutorial in Istio Security Basics.
```
What's Istio Security Basics? Link?

liminw replied:
We plan to add a security CUJ. We can link to that doc once it's added.

geeknoid commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -1,29 +1,28 @@
-  [quick start](/docs/setup/kubernetes/quick-start/).
-  Note that authentication should be enabled at step 5 in the
-  [installation steps](/docs/setup/kubernetes/quick-start/#installation-steps).
+The tasks in this guide assume that you:
```
The activities in this task assume

yangminzhu replied:
Done.

geeknoid commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -1,29 +1,28 @@
+* Deploy the [Bookinfo](/docs/examples/bookinfo/) sample application.
 
-*   In this task, we will enable access control based on Service Accounts, which are cryptographically authenticated in the Istio mesh.
+* In this task, we will enable access control based on Service Accounts, which are cryptographically authenticated in the Istio mesh.
```
Istio mesh -> mesh

yangminzhu replied:
Done.

geeknoid commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -34,57 +33,69 @@ microservices running under them.
+explicitly define access control policy to grant access to any service.
 
-> There may be delay due to caching on browser and Istio proxy.
+> There may be delay due to caching on browser and policy delayed delivered to Istio proxy.
```
Replace with

There may be some delays due to caching and other propagation overhead.

yangminzhu replied:
Done.

geeknoid commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -137,25 +148,25 @@ servicerolebinding "bind-service-viewer" created
 with "Book Details" section in the lower left part and "Book Reviews" section in the lower right part.
 
-  > There may be delay due to caching on browser and Istio proxy.
+> There may be delay due to caching on browser and policy delayed delivered to Istio proxy.
```
Replace with same sentence as above.

yangminzhu replied:
Done.

geeknoid commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -204,7 +215,7 @@ page. But there are errors `"Error fetching product details"` and `"Error fetchi
 in the following steps.
 
-> There may be delay due to caching on browser and Istio proxy.
+> There may be delay due to caching on browser and policy delayed delivered to Istio proxy.
```
Replace with above sentence.

yangminzhu replied:
Done.

geeknoid commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -204,7 +215,7 @@ page. But there are errors `"Error fetching product details"` and `"Error fetchi
-> There may be delay due to caching on browser and Istio proxy.
+> There may be delay due to caching on browser and policy delayed delivered to Istio proxy.
 
 ### Step 2. allowing "productpage" service to access "details" and "reviews" services
```
Why are the steps called out as headers, instead of just using a numbered list? These super long headings don't look good in the table of contents on the rendered web page.

yangminzhu replied:
I made it a little shorter so it should be able to be rendered in one line in most cases. I think the original idea is to make it more obvious and easier to navigate.

geeknoid commented on content/docs/tasks/security/role-based-access-control/index.md:
```diff
@@ -257,7 +268,7 @@ there is an error `"Ratings service currently unavailable"`. This is because "re
 We will show how to do that in the next step.
 
-> There may be delay due to caching on browser and Istio proxy.
+> There may be delay due to caching on browser and policy delayed delivered to Istio proxy.
```
Replace with sentence from above.

yangminzhu replied:
Done.

istio-testing commented:
@yangminzhu: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

yangminzhu commented:
Martin is OOO this week so I'll try to merge this now to make it available for testing before 1.0 release. @geeknoid, I have addressed all comments, let me know if you have any other comments.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/istio/istio.github.io/pull/1789#" title="Author self-approved">yangminzhu</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[content/OWNERS](https://github.com/istio/istio.github.io/blob/master/content/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["sdake"]} -->

istio-testing commented:
New changes are detected. LGTM label has been removed.
